suite-common/fiat-services/src/rates.ts

```diff
@@ -28,77 +35,112 @@ const getConnectFiatRatesForTimestamp = (
         },
     );
 
-export const fetchCurrentFiatRates = async (
-    ticker: TickerId,
-    currency: FiatCurrencyCode,
-): Promise<number | null | undefined> => {
-    const { success, payload } = await scheduleAction(
-        () =>
-            TrezorConnect.blockchainGetCurrentFiatRates({
-                coin: ticker.symbol,
-                token: ticker.tokenAddress,
-                currencies: [currency],
-            }),
-        { timeout: CONNECT_FETCH_TIMEOUT },
-    );
-
-    const rate = success ? payload.rates?.[currency] : null;
-
-    if (rate) return rate;
+export const fetchCurrentFiatRates = async ({
+    ticker,
+    localCurrency,
+    isElectrumBackend,
+}: fiatRatesParams): Promise<number | null | undefined> => {
+    if (isBlockbookBasedNetwork(ticker.symbol)) {
+        if (!isElectrumBackend) {
+            const { success, payload } = await scheduleAction(
+                () =>
+                    TrezorConnect.blockchainGetCurrentFiatRates({
+                        coin: ticker.symbol,
+                        token: ticker.tokenAddress,
+                        currencies: [localCurrency],
+                    }),
+                { timeout: CONNECT_FETCH_TIMEOUT },
+            );
+
+            const rate = success ? payload.rates?.[localCurrency] : null;
+
+            return rate;
+        }
+
+        const blockbookResponse = await blockbookService.fetchCurrentFiatRates(
+            'btc',
+            undefined,
+            localCurrency,
+        );
+
+        if (blockbookResponse) return blockbookResponse.rates?.[localCurrency];
+    }
 
-    const response = blockbookService.isTickerSupported(ticker)
-        ? await blockbookService.fetchCurrentFiatRates(ticker.symbol, undefined, currency)
-        : await coingeckoService.fetchCurrentFiatRates(ticker);
+    const coingeckoResponse = await coingeckoService.fetchCurrentFiatRates(ticker);
 
-    return response?.rates?.[currency];
+    return coingeckoResponse?.rates?.[localCurrency];
 };
 
-export const fetchLastWeekFiatRates = async (
-    ticker: TickerId,
-    currency: FiatCurrencyCode,
-): Promise<number | null | undefined> => {
+export const fetchLastWeekFiatRates = async ({
+    ticker,
+    localCurrency,
+    isElectrumBackend,
+}: fiatRatesParams): Promise<number | null | undefined> => {
     const weekAgoTimestamp = getUnixTime(subWeeks(new Date(), 1));
     const timestamps = [weekAgoTimestamp];
 
-    const { success, payload } = await getConnectFiatRatesForTimestamp(
-        ticker,
-        timestamps,
-        currency,
-    );
+    if (isBlockbookBasedNetwork(ticker.symbol)) {
```

Comment:
When someone is running his own custom blockbook, does he have fiat rates? @martinboehm

Reply:
He will have by default free tier for Coingecko